layout-server/src/main/java/org/finos/vuu/layoutserver/service/LayoutService.java

```diff
@@ -0,0 +1,47 @@
+package org.finos.vuu.layoutserver.service;
+
+import lombok.RequiredArgsConstructor;
+import org.finos.vuu.layoutserver.model.Layout;
+import org.finos.vuu.layoutserver.model.Metadata;
+import org.finos.vuu.layoutserver.repository.LayoutRepository;
+import org.finos.vuu.layoutserver.repository.MetadataRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class LayoutService {
+
+    private final LayoutRepository layoutRepository;
+    private final MetadataRepository metadataRepository;
+
+    public Layout getLayout(UUID id) {
+        return layoutRepository.findById(id).orElseThrow();
+    }
+
+    public List<Metadata> getMetadata() {
+        List<Metadata> metadata = new ArrayList<>();
+        layoutRepository.findAll().forEach(layout -> metadata.add(layout.getMetadata()));
+        return metadata;
+    }
+
+    @Transactional
+    public UUID createLayout(Layout layout) {
+        Metadata metadata = metadataRepository.save(layout.getMetadata());
+        metadata.setLayout(layout);
+        layout.setMetadata(metadata);
+        return layoutRepository.save(layout).getId();
+    }
```

Comment:
I thought that having `CascadeType.ALL` on the Layout entity meant you wouldn't need to do a double save?

Reply:
So did I! And yet for the life of me I couldn't get that to work. You are more than welcome to try, I'll buy you a chocolate bar if you get it to work without this double save hell.

IIRC, it was something to do with the fact that two persistence contexts were being created before the other was flushed: https://www.baeldung.com/jpa-hibernate-persistence-context.

Reply:
I can't get this to work on my side. Can you show me what you did?

Reply:
Discussed offline

Reply:
Done! [e4d2294](https://github.com/ScottLogic/finos-vuu/pull/57/commits/e4d2294edefe05736d1cc345e4c994dc7d2fa562)

Reply:
Exploring https://studygyaan.com/spring-boot/one-to-one-relationship-in-jpa

Reply:
> Exploring https://studygyaan.com/spring-boot/one-to-one-relationship-in-jpa

I tried the 3 approaches. Foreign key and join table produce the same transient exception as before. The second one produces a `attempted to assign id from null one-to-one property [org.finos.vuu.layoutserver.model.Metadata.layout` exception. I guess that's because it's attempting to use the shared ID when the ID doesn't yet exist because it's not been generated from a save.